tools/server/server-models.cpp

```diff
@@ -383,7 +391,7 @@ void server_models::unload_lru() {
         for (const auto & m : mapping) {
             if (m.second.meta.is_active()) {
                 count_active++;
-                if (m.second.meta.last_used < lru_last_used) {
+                if (!m.second.meta.pinned && m.second.meta.last_used < lru_last_used) {
```

Comment:
undefined behaviors:
- what if all models has `pin`
- what if all **loaded** models has `pin`

Reply:
I don't think violating max-models is an acceptable behavior. Throw an error instead